File: src/main/java/com/range/rpms/users/service/AuthService.java
package com.range.rpms.users.service;

import com.range.rpms.users.dto.UserLoginRequest;
import com.range.rpms.users.dto.UserRegisterRequest;
import org.springframework.stereotype.Service;

@Service
public interface AuthService {
    String login(UserLoginRequest userLoginRequest);
    String register (UserRegisterRequest userRegisterRequest);
}

File: src/main/kotlin/com/range/rpms/users/service/AuthService.kt
package com.range.rpms.users.service

import com.range.rpms.users.dto.AuthResponse
import com.range.rpms.users.dto.LoginRequest
import com.range.rpms.users.dto.RegisterRequest
import com.range.rpms.users.dto.ResetPasswordRequest

interface AuthService {
    fun login(loginRequest: LoginRequest): AuthResponse
    fun register(registerRequest: RegisterRequest): AuthResponse
    fun resetPassword(resetPasswordRequest: ResetPasswordRequest)
    fun forgotPassword(email: String)
    fun acceptTwoFactoryAuthRequest(pendingId:String,token: String): AuthResponse
    fun logout()
}